hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterHeartbeatManager.java

```diff
@@ -644,9 +674,94 @@ private void resetHeartbeats() {
     /** Remove the {@code member}'s heartbeat timestamps */
     void removeMember(MemberImpl member) {
         heartbeatFailureDetector.remove(member);
+        if (icmpParallelMode) {
+            icmpFailureDetector.remove(member);
+        }
     }
 
     void reset() {
         heartbeatFailureDetector.reset();
+        if (icmpParallelMode) {
+            icmpFailureDetector.reset();
+        }
     }
+
+    private class PingTask
+            implements Runnable {
+
+        final Member member;
+
+        PingTask(Member member) {
+            this.member = member;
+        }
+
+        public void run() {
+            try {
+                Address address = member.getAddress();
+                logger.warning(format("%s will ping %s", node.getThisAddress(), address));
+                for (int i = 0; i < MAX_PING_RETRY_COUNT; i++) {
+                    if (doPing(address, true)) {
+                        return;
+                    }
+                }
+                // host not reachable
+                String reason = format("%s could not ping %s", node.getThisAddress(), address);
+                logger.warning(reason);
+                clusterService.suspectMember(member, reason, true);
+            } catch (Throwable ignored) {
+                EmptyStatement.ignore(ignored);
+            }
+        }
+
+        boolean doPing(Address address, boolean info)
```

Comment:
_minor:_ You can pass logging level instead of a bool flag, and log via `logger.log(level, message)`.